src/main/java/com/influxdb/v3/client/internal/VectorSchemaRootConverter.java

```diff
@@ -46,94 +46,152 @@
+                return value;
             }
-        } else if (value instanceof LocalDateTime) {
-            pointValues.setTimestamp(((LocalDateTime) value).toInstant(ZoneOffset.UTC));
         }
+
+        String[] parts = metaType.split("::");
+        String valueType = parts[2];
+        if ("field".equals(valueType)) {
+            switch (metaType) {
+                case "iox::column_type::field::integer":
+                case "iox::column_type::field::uinteger":
+                    if (value instanceof Number) {
+                        return TypeCasting.toLongValue(value);
+                    } else {
+                        LOG.warning(String.format("Value %s is not an Long", value));
+                        return value;
+                    }
+                case "iox::column_type::field::float":
+                    if (value instanceof Number) {
+                        return TypeCasting.toDoubleValue(value);
+                    } else {
+                        LOG.warning(String.format("Value %s is not a Double", value));
+                        return value;
+                    }
+                case "iox::column_type::field::string":
+                    if (value instanceof Text || value instanceof String) {
+                        return TypeCasting.toStringValue(value);
+                    } else {
+                        LOG.warning(String.format("Value %s is not a String", value));
+                        return value;
+                    }
+                case "iox::column_type::field::boolean":
+                    if (value instanceof Boolean) {
+                        return value;
+                    } else {
+                        LOG.warning(String.format("Value %s is not a Boolean", value));
+                        return value;
+                    }
+                default:
+                    return value;
+            }
+        } else if ("timestamp".equals(valueType) || Objects.equals(fieldName, "time")) {
+            return NanosecondConverter.getTimestampNano(value, field);
+        } else {
+            return TypeCasting.toStringValue(value);
+        }
+    }
+
+    /**
+     * Get array of values from VectorSchemaRoot.
+     *
+     * @param vector    The data return from InfluxDB.
+     * @param rowNumber The row number of data
+     * @return  An array of Objects represents a row of data
+     */
+    public Object[] getArrayObjectFromVectorSchemaRoot(@Nonnull final VectorSchemaRoot vector, final int rowNumber) {
+        int columnSize = vector.getFieldVectors().size();
+        var row = new Object[columnSize];
+        for (int i = 0; i < columnSize; i++) {
+            FieldVector fieldVector = vector.getVector(i);
+            row[i] = getMappedValue(
+                    fieldVector.getField(),
+                    fieldVector.getObject(rowNumber)
+            );
+        }
+
+        return row;
```

Comment:
Just little change for better future maintenance:

```suggestion
        List<FieldVector> fieldVectors = vector.getFieldVectors();
        int columnSize = fieldVectors.size();
        var row = new Object[columnSize];
        for (int i = 0; i < columnSize; i++) {
            FieldVector fieldVector = fieldVectors.get(i);
            row[i] = getMappedValue(
                    fieldVector.getField(),
                    fieldVector.getObject(rowNumber)
            );
        }
        return row;
```